src/parser/grammar.y

```diff
@@ -216,22 +217,54 @@ link(A) ::= LEFT_ARROW edge(B) DASH . {
 %type edge {AST_LinkEntity*}
 // Empty edge []
 edge(A) ::= LEFT_BRACKET properties(B) RIGHT_BRACKET . { 
-	A = New_AST_LinkEntity(NULL, NULL, B, N_DIR_UNKNOWN);
+	A = New_AST_LinkEntity(NULL, NULL, B, N_DIR_UNKNOWN, NULL);
 }
 
 // Edge with alias [alias]
 edge(A) ::= LEFT_BRACKET UQSTRING(B) properties(C) RIGHT_BRACKET . { 
-	A = New_AST_LinkEntity(B.strval, NULL, C, N_DIR_UNKNOWN);
+	A = New_AST_LinkEntity(B.strval, NULL, C, N_DIR_UNKNOWN, NULL);
 }
 
 // Edge with label [:label]
-edge(A) ::= LEFT_BRACKET COLON UQSTRING(B) properties(C) RIGHT_BRACKET . { 
-	A = New_AST_LinkEntity(NULL, B.strval, C, N_DIR_UNKNOWN);
+edge(A) ::= LEFT_BRACKET COLON UQSTRING(B) edgeLength(C) properties(D) RIGHT_BRACKET . { 
+	A = New_AST_LinkEntity(NULL, B.strval, D, N_DIR_UNKNOWN, C);
 }
 
 // Edge with alias and label [alias:label]
-edge(A) ::= LEFT_BRACKET UQSTRING(B) COLON UQSTRING(C) properties(D) RIGHT_BRACKET . { 
-	A = New_AST_LinkEntity(B.strval, C.strval, D, N_DIR_UNKNOWN);
+edge(A) ::= LEFT_BRACKET UQSTRING(B) COLON UQSTRING(C) edgeLength(D) properties(E) RIGHT_BRACKET . { 
+	A = New_AST_LinkEntity(B.strval, C.strval, E, N_DIR_UNKNOWN, D);
+}
+
+// Edge with length [alias:*1..3]
```

Comment:
Cypher doesn't support edges that have aliases but no labels in general (a change we implemented about a month ago, I think), which remains true with length specifiers:
```
MATCH (a)-[alias:*]->(b) RETURN b
> Invalid input '*': expected whitespace or a rel type name
```
We should, however, support edges with labels and length specifications, like:
`MATCH (a)-[:KNOWS*..2]->(b) RETURN b`

The Cypher Query Language Reference (version 9, pages 11-12) doesn't explicitly mention edges with labels, aliases, and length specifications, but they are supported in Neo.
`MATCH (a)-[alias:KNOWS*]->(b) RETURN b`

Reply:
You're correct, although I don't see a reason why we shouldn't support this
I think its quite powerful to ask (A)-[:*1..20]->(B)

Reply:
@swilly22 As a minor clarification, `(A)-[*1..20]->(B)` (returning all matching paths without providing an alias to refer to those paths in other parts of the query) is totally canonical syntax, and we should support it!

Cypher does not explicitly support `(A)-[alias:*1..20]->(B)`, and - conjecturing - I suspect that is because it's proven computationally challenging to merge the properties of all matching paths into a single edge alias that can be filtered, returned, and so forth. I can't think of any syntactical reason for this constraint other than that it taxes the system, especially given that Cypher provides support for multiple explicit edge labels with syntax like `(a)-[:KNOWS|HAS_MET)->(b)`.

I have no objection to going beyond the language specification, but providing support for edge aliases that don't correspond to specific labels will introduce a fair few cases we'll need to add tooling for. As an example, if one wants to filter on a property found along a path, we'll need to find and track the properties of _all_ paths matching the query, as opposed to the masking approach suggested in this PR, which at each hop only considers edges that can modify the paths being built. I'm specifically referring to the `_VarLenTraverseOpHop` logic that builds and employs a mask to ignore traversals that, on a boolean level, have been made redundant, but I expect that is not the only edge case we'd need to compensate for.

This is just food for thought - I'm on board with providing more functionality whenever someone might find it useful! Let me know what you think.